### PR TITLE
Document excluding pages and directories from build

### PR DESCRIPTION
**Pull request by zadeviggers**
Documents how to exclude pages or directories from being built by prefixing their name with an underscore.

### CODE REVIEW

File: src/pages/en/core-concepts/routing.md
Comment (sarah11918) [inline src/pages/en/core-concepts/routing.md:97]:
```suggestion

## Excluding pages

You can exclude pages, or even whole directories from being built by prefixing their names with an underscore (`_`).

This allows you to co-locate tests, utilities, and components with their related pages and prevent them from being built into html files and placed into the `dist/` directory.

```
└── pages/
   ├── posts/
   │   ├── post1.md
   │   ├── SomeComponent.astro
   │   └── utils.js
   ├── _hidden-directory/
   |   ├── page1.md
   |   └── page2.md
   ├── _hidden-page.astro
   └── index.astro
   
```

In the above example, only `src/pages/index.astro` and `src/pages/post1.md` will be built as page routes and html files.
```
What do you think about an example like this, that shows the kinds of files you were talking about? (Feel free to change it to make it more realistic.) Just a thought.

Since this is the page on routing, I did want to slip in the phrase "page route" somewhere in this description, too.

File: src/pages/en/core-concepts/routing.md
Comment (aFuzzyBear) [inline src/pages/en/core-concepts/routing.md:103]: I would prefer if this was a tree overview of the `src/pages/` directory, for it would summarise the concepts that are encapsulated into the page and would fit nicely into this section.

```tree
src/pages/
 ├──  _tests/       #Private and excluded from build
  |       └──test1.astro
 ├──  blog/        
  |       ├── [...posts].astro
  |       └── index.astro
 └──  index.astro
 ```

**Review by aFuzzyBear — DISMISSED** [review]
I made a few comments, otherwise I think this would be a great addition to the page
Great initiative @zadeviggers  👊

File: src/pages/en/core-concepts/routing.md
Comment (aFuzzyBear) [reply src/pages/en/core-concepts/routing.md:97]: Ah Im sorry buddy, I didnt see your comment, I posted my response below,
It's funny how we both had the same idea... 😂

Comment (sarah11918) [reply src/pages/en/core-concepts/routing.md:97]: I prefer keeping this as "Excluding pages" because it's the goal: stop things from being built. It's also about excluding files that are components etc., not just about making pages private.

File: src/pages/en/core-concepts/routing.md
Comment (sarah11918) [reply src/pages/en/core-concepts/routing.md:101]: I would personally use *my* wording, in my comment above.  ;)  I think this could benefit from being shorter, not longer.

File: src/pages/en/core-concepts/routing.md
Comment (sarah11918) [reply src/pages/en/core-concepts/routing.md:103]: Agree with the tree, and now feel vindicated since @aFuzzyBear had the same thought! Go Team!

@zadeviggers - Can you work with my tree above (I think it shows a bunch of different types of things to be excluded, which is the point) and adjust it to be more realistic, if you need to?

File: src/pages/en/core-concepts/routing.md
Comment (zadeviggers) [reply src/pages/en/core-concepts/routing.md:97]: I agree with Sarah on this one

File: src/pages/en/core-concepts/routing.md
Comment (zadeviggers) [reply src/pages/en/core-concepts/routing.md:103]: Okay, will do!

File: src/pages/en/core-concepts/routing.md
Comment (zadeviggers) [reply src/pages/en/core-concepts/routing.md:97]: Alright, I added a slightly modified version of this to the section under an example heading.

**Review by sarah11918 — APPROVED** [review]

File: src/pages/en/core-concepts/routing.md
Comment (sarah11918) [inline src/pages/en/core-concepts/routing.md:103]:
```suggestion
#### Example: Excluding pages
```

What do you think? If we add one more to the header, then it doesn't show up in the sidebar. I don't think having it show up as its own sub-heading in the sidebar adds anything.

**Comment by sarah11918** [comment]
I think this looks good! Short and sweet! Just left one comment re: the header for the example.  :)

File: src/pages/en/core-concepts/routing.md
Comment (zadeviggers) [reply src/pages/en/core-concepts/routing.md:103]: I agree, but then there's a semantics issue because we're skipping a heading level.
Maybe removing the heading altogether would be better? That would make it less consistent with the rest of the page, but because it's not a very large section it might not be noticeable.

Comment (sarah11918) [reply src/pages/en/core-concepts/routing.md:103]: Sure, I'd be fine with removing it entirely!

**Comment by sarah11918** [comment]
(FYI @zadeviggers , I'm just waiting to merge this until a larger PR on that Routing page is merged. Then, this will just be a tiny, non-conflicting addition to it.)  :)

**Comment by FredKSchott** [comment]
This is good documentation but I'm actually not sure if we _meant_ for this feature to exist. If you're alright with sitting tight, I'd like to run this by our maintainers group and get their +1/-1 before we document.

**Comment by 5310** [comment]
I really hope this feature _continues_ to exist, or my projects will become _significantly_ more work than what I already have to do because I can't co-locate static assets with my content :heavy_plus_sign: :infinity: 

In fact, please consider adding pattern-based allow/deny lists for Astro to process someday in the future :blush:

**Comment by sarah11918** [comment]
Poor @zadeviggers still waiting to hear whether he found an undocumented feature, or an easter egg, or what! Just wanted to let you know that I haven't forgotten about you, and I have added a label to prompt core to weigh in on this.

**Review by delucis — APPROVED** [review]

**Comment by tony-sull** [comment]
Lets bring this up on an RFC call!  The feature has come up a few times before but was never decided on, since it already technically works the question really is just whether it's officially supported 👍

**Comment by sarah11918** [comment]
@tony-sull Should @zadeviggers make an official RFC proposal for this on withastro/rfc ?

One "gotcha" coming up: I suspect we're going to move this Routing page back into "Learn" (and out of API). We've already worked so hard to get the wording/diagram *just right*, but so much has changed and I can see there are already going to be some conflicts to resolve, even before we move the page... 

I'm wondering whether it makes sense for @zadeviggers to capture these exact edits and stash them somewhere (in an RFC?!) and we close this PR since it still seems no where near getting the go-ahead to merge. 

That way, the rest of the site can change, this page can move to an entirely new location... and maybe as a result of the RFC the team decides that they want to implement this slightly differently anyway, so we're waiting on those decisions before we know what to document?

What does everyone think?

**Comment by delucis** [comment]
Does @zadeviggers _want_ to do that? It does seem like it might be a path to grabbing some of core's attention and getting a decision. But don't know if this feature is important enough to you that you would want to champion it in this way.

**Comment by sarah11918** [comment]
Well, THAT's a good point, too! Sorry, @zadeviggers , didn't mean to be putting RFCs in your mouth!  ;)

Even if we end up closing this PR, the info is still "stashed" here, should we need it again. I just think this feels no where near close to being an actual thing to document yet, and I'm trying to think about how to proceed in light of that. Open to other ideas, too!

A problem with just documenting it as it currently happens to work (as I might tend to do for convenience) without a clear committment of future support is that people might build their entire projects around it not realizing that it's not a core/key thing and no one is working to ensure that it supported.

**Comment by delucis** [comment]
I think it makes sense to at least convert this to a draft. @zadeviggers Would you like me to take on writing up an RFC? I think this feature or something like it would be super helpful and given it already exists I think it would be good to get a quick-ish 👍/👎 from the team.

**Comment by zadeviggers** [comment]
> @zadeviggers Would you like me to take on writing up an RFC? I think this feature or something like it would be super helpful and given it already exists I think it would be good to get a quick-ish 👍/👎 from the team.

Of course! Feel free to do whatever you want, no need to ask me first.

File: src/pages/en/core-concepts/routing.md
Comment (sarah11918) [inline src/pages/en/core-concepts/routing.md:196]:
```suggestion
This allows you to create private pages, and also to co-locate tests, utilities, and components with their related pages, preventing them from being built into `.html` files and placed into the `dist/` directory.

In this example, only `src/pages/index.astro` and `src/pages/post1.md` will be built as page routes and HTML files.
```

**Comment by chenxsan** [comment]
> 



> I think it makes sense to at least convert this to a draft.
> 
> @zadeviggers Would you like me to take on writing up an RFC? I think this feature or something like it would be super helpful and given it already exists I think it would be good to get a quick-ish 👍/👎 from the team.

> 

Can you point me to the location of this RFC? Failed to find it in https://github.com/withastro/rfcs

I found this excluding feature to be confusing when used alongside `Astro.glob`, since there's no pages generated when their names are prefixed with underscore, but you still have them listed in `Astro.glob`'s result, which means you still have to filter them with code like:

```js
allPosts
      .filter((post) => {
        return !path.basename(post.file).startsWith('_') // only works for files prefixed with _, not directories prefixed with _
      })
```

**Pull request merged**
